src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -1043,6 +1043,11 @@ private void EndTagBlock(Stack<Tuple<HtmlSymbol, SourceLocation>> tags, bool com
             if (!Context.DesignTimeMode)
             {
                 AcceptWhile(HtmlSymbolType.WhiteSpace);
+                if (Context.LastSpan.Kind == SpanKind.Transition)
+                {
+                    // Ignore the whitespace at the end of a <text> tag.
+                    Span.ClearSymbols();
```

Comment:
Can't clear symbols because it'd destroy the tree without accounting for the symbols. The space after `<text>` should be qualified as code.